### PR TITLE
fix(developer): trim new words entered into the wordlist editor

### LINKED ISSUE #12668: bug(developer): A new "Add word" row is created by pressing the spacebar.
### Describe the bug

The "Add word" row is created when pressing the spacebar key.

### Reproduce the bug

Steps to reproduce:
1. I installed the keymandeveloper-18.0.138-alpha build in Windows 10.
2. Created a new "Lexical Model" project.
3. Navigate to the "Models" tab and click the "*.model.ts" link button.
4. Navigate to the *.tsv file tab.
5. Click in the "Add word..." box.
6. Press the "Spacebar" key.
Actual Results: Here, the "Add word..." row was added.
Please take a look at the screenshot (AddWord_SpaceBar.png).

### Expected behavior

_No response_

### Related issues

_No response_

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [ ] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [X] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

 keymandeveloper-18.0.138-alpha

### Operating system

Windows 10 OS

### Device

_No response_

### Target application

_No response_

### Browser

_No response_

### Keyboard name

_No response_

### Keyboard version

_No response_

### Language name

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes: #12668

# User Testing

* TEST_WORDLIST_EDITOR: Open the Wordlist Editor and try entering spaces into new rows in the editor. Upon exiting the cell, the text should be trimmed, and if only whitespace was present on the last row, it should return to being a 'Add word...' entry. Other words entered should also be trimmed of leading and following whitespace.

### CODE REVIEW

**Review by ermshiperete — APPROVED** [review]

File: developer/src/tike/child/Keyman.Developer.UI.UframeWordlistEditor.pas [hunk 1]
```diff
@@ -313,6 +315,25 @@ procedure TframeWordlistEditor.gridWordlistDrawCell(Sender: TObject; ACol,
...
+  if gridWordlist.Col = 0 then
+  begin
+    // The value is trimmed in the backing data but trim it here also when
+    // we exit a cell, so the visible text matches
+    Value := Trim(gridWordlist.Cells[gridWordlist.Col, gridWordlist.Row]);
+    gridWordlist.Cells[gridWordlist.Col, gridWordlist.Row] := Value;
+    if (gridWordlist.Row = gridWordlist.RowCount - 1) and (Value = '') then
+    begin
+      // And if the last row was just whitespace, show the hint text again
+      FillGridNewRow;
```
Comment (jahorton) [inline developer/src/tike/child/Keyman.Developer.UI.UframeWordlistEditor.pas:332]: Do we check that the _new_ last row is also not just whitespace?  Is that "edge case" enough that we don't care?  Or is there some mechanism I'm not seeing that handles such a case?

**Comment by Meng-Heng** [comment]
# Test Specs
1. Keyman Developer 19.0.281-alpha-test-16538

# Test Results

* **TEST_WORDLIST_EDITOR (PASSED):**
1. Launch Keyman Developer -> Open a wordlist project
2. Enter new words `algelo` + space
3. Click out of the cell
4. VERIFIED: The space is trimmed
5. Enter Spaces before and after the words
6. VERIFIED: The spaces are trimmed 
7. Add just spaces to the last row
8. VERIFIED: The last row shows "Add word..."

File: developer/src/tike/child/Keyman.Developer.UI.UframeWordlistEditor.pas (hunk 1 above)
Comment (mcdurdin) [reply developer/src/tike/child/Keyman.Developer.UI.UframeWordlistEditor.pas:332]: This doesn't add an additional row; rather it causes the 'Add word...' text to appear in the last row again.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 19.0.286-alpha](https://keyman.com/downloads/releases/alpha/19.0.286)
